jax/experimental/jax2tf/jax2tf.py

```diff
@@ -126,6 +126,95 @@ def add_unit(v: TfValOrUnit, aval: core.AbstractValue):
     return core.unit if aval is core.abstract_unit else v
   return util.safe_map(add_unit, vals, avals)
 
+
+Limitation = NamedTuple("Limitation", [ ("PrimitiveName", str)
+                                      , ("ErrorType", str)
+                                      , ("ErrorString", str)
+                                      , ("Devices", Tuple[str,...])
+                                      ])
+
+def categorize(prim: core.Primitive, *args: TfVal, **kwargs) \
+    -> List[Limitation]:
+  """
+  Given a primitive and a set of parameters one would like to pass to it,
+  categorize identifies the potential limitations the call would encounter when
+  converted to TF through jax2tf.
+
+  Args:
+    prim: the primitive to call.
+    args: the arguments to pass to prim.
+    kwargs: the keyword arguments to pass to prim.
+
+  Returns:
+    A list of limitations
+  """
+  limitations: List[Limitation] = []
+  all_devices = ["CPU", "GPU", "TPU"]
+
+  def _report_failure(error_type: str, msg: str,
```

Comment:
I think it may make sense to actually run this as it is intended to generate the report, and to include the report in the PR.

I would also change one of the tests with `min` and `max` to see what would it take for this solution to be used to control the test failure. Ideally, we would see that we can remove this conditional from the test.

Reply:
Addressed both of these in the next commits.
I also added a flag for test generation without file generation, as I assume the tests may fail in Google otherwise.